jax/_src/lax/control_flow.py

```diff
@@ -383,43 +383,77 @@ def _while_loop_batching_rule(args, dims, axis_name, main_type,
   size, = {x.shape[d] for x, d in zip(args, dims) if d is not batching.not_mapped}
   orig_batched = [d is not batching.not_mapped for d in dims]
   cconst_bat, bconst_bat, init_bat = split_list(orig_batched, [cond_nconsts, body_nconsts])
+  cconsts, bconsts, init = split_list(args, [cond_nconsts, body_nconsts])
+  cconst_dims, bconst_dims, init_dims = split_list(dims, [cond_nconsts, body_nconsts])
 
+  carry_bat = init_bat
   # Fixpoint computation of which carry are batched: either
   # batched from init, or the carry out is batched. Each iteration promotes
   # at least one carry to batched. We need at most len(carry) iterations,
   # but we need one last iteration to prepare the jaxpr based on the final
   # carry_bat.
-  carry_bat = init_bat
   for _ in range(1 + len(carry_bat)):
-    batched = bconst_bat + carry_bat
-    body_jaxpr_batched, carry_bat_out = batching.batch_jaxpr(
-        body_jaxpr, size, batched, instantiate=carry_bat,
-        axis_name=axis_name, main_type=main_type)
-    cond_jaxpr_batched, (pred_bat,) = batching.batch_jaxpr(
-        cond_jaxpr, size, cconst_bat + carry_bat,
-        instantiate=bool(cond_jaxpr.out_avals[0].shape),
+    _, carry_bat_out = batching.batch_jaxpr(
+        body_jaxpr, size, bconst_bat + carry_bat, instantiate=False,
         axis_name=axis_name, main_type=main_type)
-    carry_bat_out = _map(partial(operator.or_, pred_bat), carry_bat_out)
-    if carry_bat_out == carry_bat:
+    if carry_bat == carry_bat_out:
       break
-    else:
-      carry_bat = _map(operator.or_, carry_bat, carry_bat_out)
+    carry_bat = safe_map(operator.or_, carry_bat, carry_bat_out)
   else:
-    assert False, "Fixpoint not reached"
+    raise ValueError("Fixpoint not reached")
```

Comment:
Can you please make this `AssertionError`? `ValueError` is for invalid argument values, but this looks more like an internal logic error

Reply:
Oops, forgot to change this back. Thanks for the catch.